src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -493,3 +502,18 @@
       return(list(error = e))
    })
 })
+
+.rs.addFunction("preview_data_import_async", function(jsonData)
+{
+   tryCatch({
+      params = jsonlite::fromJSON(jsonData)
```

Comment:
Oh, I see now that this is getting executed in a vanilla R process. You could pass the params as vanilla strings, and for the output you could have the session give the async process the name of a .RData file to write the result to, which you can read and delete when the process terminates.